core/src/Instance/B2CAuthority.cs

```diff
@@ -63,5 +65,14 @@ internal override string GetTenantId()
         {
             return new Uri(CanonicalAuthority).Segments[2].TrimEnd('/');
         }
+
+        internal override void UpdateTenantId(string tenantId)
+        {
+            Uri authorityUri = new Uri(CanonicalAuthority);
+            var segments = authorityUri.Segments;
+
+            CanonicalAuthority = string.Format(CultureInfo.InvariantCulture, B2CCanonicalAuthorityTemplate, 
+                authorityUri.Authority, segments[1].TrimEnd('/'), tenantId, segments[3].TrimEnd('/'));
```

Comment:
It's pretty hard to read segments[1], segments[3] etc. - maybe create local variables or give an example

Reply:
good point - will change